File: c++/Product.cpp
#include <iostream>
#include <string>

using namespace std;

class Product
{
    private:
        /* atribut private */
        int price;
        string idProduct;
    
    public:
        // constructor
        Product(){
        }

        // constructor dengan parameter
        Product(int p, string idP){
            /* Mengisi class Product */
            price = p;
            idProduct = idP;
        }

        //membuat getter dan setter
        void setPrice(int p){
            price = p;
        }

        int getPrice(){
            return price;
        }

        void setIdProduct(string idP){
            idProduct = idP;
        }

        string getIdProduct(){
            return idProduct;
        }

        //prosedur untuk memnampilkan atribut kelas
        void showProduct(){
            cout << "ID Product    : " << idProduct << endl;
            cout << "Price         : Rp. " << price << endl;
        }



        ~Product();
};

File: java/Product.java
public class Product {

    private int price;
    private String idProduct;

    public Product(){
    }

    public Product(int p, String idP){
        this.price = p;
        this.idProduct = idP;
    }

    public int getPrice() {
        return price;
    }

    public String getIdProduct() {
        return idProduct;
    }

    public void setIdProduct(String idProduct) {
        this.idProduct = idProduct;
    }

    public void setPrice(int price) {
        this.price = price;
    }

    public void showProduct() {
        System.out.println("ID Product    : " + idProduct);
        System.out.println("Price         : Rp. " + price);
    }
}